j5/backends/hardware/sr/v4/serial/power_board.py

```diff
@@ -0,0 +1,354 @@
+
+        :returns: Response, if any.
+        """
+        request_data = command.encode("ascii") + b'\n'  # TODO: Handle emoji lol
+
+        with self._lock:
+            try:
+                bytes_written = self._serial.write(request_data)
+                if len(request_data) != bytes_written:
+                    raise CommunicationError(
+                        "Mismatch in command bytes written to serial interface.",
+                    )
+            except SerialTimeoutException as e:
+                raise CommunicationError(f"Serial Timeout Error: {e}") from e
+            except SerialException as e:
+                raise CommunicationError(f"Serial Error: {e}") from e
+
+            response = self.read_serial_line()  # TODO: Handle decoding errors (Is this a bug elsewhere in j5? (yes))
+
+        if response[:4] == "NACK":
+            _, error_string = response.split(":", 1)
+            raise CommunicationError(f"Power Board returned an error: {error_string}")
+
+        expects_response = command[-1:] == "?"
+        if expects_response:
+            return response
+        elif response == "ACK":
+            return None
+        else:
+            raise CommunicationError(f"Expected ACK from Power Board, but got: {response}")
+
+    def request_with_response(self, command: str) -> str:
+        if not command.endswith('?'):
+            raise ValueError(f"The provided command does not expect a response: {command}")
+
+        response = self.request(command)
+        if response is not None:
+            return response
+        else:
+            raise CommunicationError("Power board responded with ACK but expected data.")
+
+    # TODO: Can we cache this?
+    def get_identity(self) -> BoardIdentity:
+        """
+        Get the board identity information.
+
+        :raises CommunicationError: the identity response did not match the expected format.
+        :returns: A tuple of board identity information
+        """
+        response = self.request("*IDN?")  # TODO: Handle None here?
+        parts = response.split(":", 4)
+        if len(parts) != 4:
+            raise CommunicationError(f"Identify response did not match format: {response}")
+        else:
+            return BoardIdentity(*parts)
+
+    def reset_board(self) -> None:
+        self.request("*RESET")
+
+    @property
+    def firmware_version(self) -> str:
+        """
+        The firmware version reported by the board.
+
+        :returns: firmware version reported by the board, if any.
+        """
+        identity = self.get_identity()
+        return identity.software_version
+
+    def get_power_output_enabled(self, identifier: int) -> bool:
+        """
+        Get whether a power output is enabled.
+
+        :param identifier: power output to fetch status of.
+        :returns: status of the power output.
+        :raises ValueError: Invalid power output identifier.
+        """
+        if identifier in range(6):
```

Comment:
There are now 7 outputs, the 7th (identifier/index 6) being the 5V output as this is no longer used by the brain. This output supports the same functionality as the others so this can just be extended.

Reply:
827cc89

Reply:
It's really not that simple sadly. We need to update the board too to support this and turn on the 5V output.

Reply:
We can do this after #814, for now I'd suggest we don't support the 5V output in this PR.

Reply:
I dropped the commit that changed it to 7. We can deal with this separately.

Reply:
Note that this will prevent the 5 volt output from being enabled.

Reply:
The frontend components don't support the 5V output yet, so we will implement it later. Also blocked on #814

Reply:
We can do this once #817 is merged